slurm/src/slurmcc/cost.py

```diff
@@ -0,0 +1,256 @@
+            log.error(e.strerror)
+            raise
+        default_output_fmt = "jobid,user,account,cluster,partition,ncpus,nnodes,submit,start,end,elapsedraw,state"
+        default_input_fmt = default_output_fmt + ",admincomment"
+        self.options = "--allusers --duplicates --parsable2 --allocations --noheader"
+        if fmt:
+            req_fmt = fmt.split(",")
+            in_fmt = default_input_fmt.split(",")
+            for f in req_fmt:
+                if f not in in_fmt:
+                    in_fmt.append(f)
+            self.output_format = ",".join(req_fmt)
+            self.input_format = ",".join(in_fmt)
+        else:
+            self.output_format = default_output_fmt
+            self.input_format = default_input_fmt
+
+        self.in_fmt_t = namedtuple('in_fmt_t', self.input_format)
+        self.slurm_fmt_t = namedtuple('slurm_fmt_t', self.output_format)
+        self.c_fmt_t = namedtuple('c_fmt_t', ['cost'])
+
+    def _construct_command(self) -> str:
+
+        args = f"{self.sacct} {self.options} " \
+                f"-M {self.cluster} "\
+                f"--start={self.start} " \
+                f"--end={self.end} -o "\
+                f"{self.input_format}"
+        return args
+
+    def use_cache(self, filename) -> bool:
+        return False
+
+    def get_queue_rec_file(self) -> str:
+        return os.path.join(self.cache, f"queue.out")
+
+    def get_job_rec_file(self) -> str:
+        return os.path.join(self.cache, f"sacct-{self.start}-{self.end}.out")
+
+    def get_queue_records(self) -> str:
+
+        _queue_rec_file = self.get_queue_rec_file()
+        if self.use_cache(_queue_rec_file):
+            return _queue_rec_file
+
+        cmd = f"{self.squeue} --json"
+        with open(_queue_rec_file, 'w') as fp:
+            output = run_command(cmd, stdout=fp)
+            if output.returncode:
+                log.error("could not read slurm queue")
+        return _queue_rec_file
+
+    def process_queue(self) -> dict:
+        running_jobs = {}
+        queue_rec = self.get_queue_records()
+        with open(queue_rec, 'r') as fp:
+            data = json.load(fp)
+
+        for job in data['jobs']:
+            if job['job_state'] != 'RUNNING' and job['job_state'] != 'CONFIGURING':
+                continue
+            job_id = job['job_id']
+            if job['admin_comment']:
+                running_jobs[job_id] = job['admin_comment']
+        return running_jobs
+
+    def fetch_job_records(self) -> str:
+
+        _job_rec_file = self.get_job_rec_file()
+        if self.use_cache(_job_rec_file):
+            return _job_rec_file
+        cmd = self._construct_command()
+        with open(_job_rec_file, 'w') as fp:
+            output = run_command(cmd, stdout=fp)
+            if output.returncode:
+                log.error("Could not fetch slurm records")
+        return _job_rec_file
+
+    def parse_admincomment(self, comment: str):
+
+        return json.loads(comment)
```

Comment:
I recommend using a helper function here. You will definitely get malformed json eventually, and you want to give as much info to the user on where this is happening. So do something that tries to parse it, logs the invalid comment, and then raises an exception. Anyone who calls this func can then capture that error and log which specific comment caused this.